standard/types.md

```diff
@@ -912,39 +912,83 @@ A diagnostic can be produced when a variable ([§9.2.1](variables.md#921-general
+> In the previous example, a compiler may decide that after the statement `int length = p.Length;`, the null-state of `p` is not-null. If it were null, that statement would have thrown a `NullReferenceException`. This is similar to the behavior if the code had been preceded by `if (p == null) throw NullReferenceException();` except that the code as written may produce a warning, the purpose of which is to warn that an exception may be thrown implicitly. *end example*
 
-Later in the method, the code checks that `s` is not a null reference. The null-state of `s` can change to maybe null after the null-checked block closes. The compiler can infer that `s` is maybe null because the code was written to assume that it might have been null. Generally, when the code contains a null check, the compiler may infer that the value might have been null.*end example*
-<!-- markdownlint-disable MD028 -->
+Later in the method, the code checks that `s` is not a null reference. The null-state of `s` can change to maybe null after the null-checked block closes. A compiler can infer that `s` is maybe null because the code was written to assume that it might have been null. Generally, when the code contains a null check, a compiler may infer that the value might have been null:
 
-<!-- markdownlint-enable MD028 -->
-> *Example*: The compiler can treat a property ([§15.7](classes.md#157-properties)) as either a variable with state, or as independent get and set accessors ([§15.7.3](classes.md#1573-accessors)). In other words, a compiler can choose whether writing to a property changes the null state of reading the property, or if reading a property changes the null state of that property.
+> *Example*: Each of the following expressions include some form of a null check. The null-state of `o` can change from not null to maybe null after each of these statements:
 >
-> <!-- Example: {template:"standalone-console", name:"NullPropertyAnalysis"} -->
+> <!-- Example: {template:"code-in-class-lib", name:"NullChecks", expectedWarnings:["CS8602", "CS8602", "CS8602"]} -->
+> ```csharp
+> #nullable enable
+> public void M(string s)
+> {
+>     int length = s.Length; // No warning. s is not null
+>
+>     _ = s == null; // Null check by testing equality. The null state of s is maybe null
+>     length = s.Length; // Warning, and changes the null state of s to not null
+>
+>     _ = s?.Length; // The ?. is a null check and changes the null state of s to maybe null
+>     if (s.Length > 4) // Warning. Changes null state of s to not null
+>     {
+>         _ = s?[4]; // ?[] is a null check and changes the null state of s to maybe null
+>         _ = s.Length; // Warning. s is maybe null
+>     }
+> }
+> ```
+
+Both auto-property and field-like event declarations make use of a compiler generated backing field. Null state analysis may infer that assignment to the event or property is an assignment to a compiler generated backing field.
```

Comment:
Uber nitpick :D
```suggestion
Both auto-property and field-like event declarations make use of a compiler-generated backing field. Null state analysis may infer that assignment to the event or property is an assignment to a compiler generated backing field.
```

Reply:
Also "auto-property" vs "auto property" but that one is likely a separate item if it's even palatable to consider.